pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java

```diff
@@ -368,4 +368,22 @@
      * then the ack timeout will be set to 30000 millisecond
      */
     ConsumerBuilder<T> deadLetterPolicy(DeadLetterPolicy deadLetterPolicy);
+
+    /**
+     * If enabled, the consumer will auto subscribe for partitions increasement.
+     * This is only for partitioned consumer and multi-topics consumer.
+     *
+     * @param autoUpdate
+     *            whether to auto update partition increasement
+     */
+    ConsumerBuilder<T> trackPartitionUpdate(boolean autoUpdate);
```

Comment:
More than "track", I think the operation here should be more explicit in that it will not only "track" but automatically start producing/consuming from the new partitions.